tensorflow/lite/micro/kernels/ceva/fully_connected.cc

```diff
@@ -0,0 +1,257 @@
+
+  auto* data = static_cast<OpDataFullyConnected*>(node->user_data);
+
+  const auto params =
+      static_cast<const TfLiteFullyConnectedParams*>(node->builtin_data);
+
+  const TfLiteTensor* input =
+      GetInput(context, node, kFullyConnectedInputTensor);
+  const TfLiteTensor* filter =
+      GetInput(context, node, kFullyConnectedWeightsTensor);
+  const TfLiteTensor* bias =
+      GetOptionalInputTensor(context, node, kFullyConnectedBiasTensor);
+  TfLiteTensor* output = GetOutput(context, node, kFullyConnectedOutputTensor);
+
+  TF_LITE_ENSURE_TYPES_EQ(context, input->type, output->type);
+  TF_LITE_ENSURE_MSG(context, input->type == filter->type,
+                     "Hybrid models are not supported on TFLite Micro.");
+
+  return CalculateOpDataFullyConnected(context, params->activation, input->type,
+                                       input, filter, bias, output, data);
+}
+
+TfLiteStatus EvalQuantizedInt8CEVA(TfLiteContext* context, TfLiteNode* node,
+                                   const OpDataFullyConnected& data,
+                                   const TfLiteEvalTensor* input,
+                                   const TfLiteEvalTensor* filter,
+                                   const TfLiteEvalTensor* bias,
+                                   TfLiteEvalTensor* output) {
+  tflite::FullyConnectedParams op_params = FullyConnectedParamsQuantized(data);
+
+  int input_shape_DimensionsCount =
+      tflite::micro::GetTensorShape(input).DimensionsCount();
+  int weights_shape_DimensionsCount =
+      tflite::micro::GetTensorShape(filter).DimensionsCount();
+  int* weights_shape_DimsData =
+      const_cast<int*>(tflite::micro::GetTensorShape(filter).DimsData());
+  int bias_shape_DimensionsCount =
+      tflite::micro::GetTensorShape(bias).DimensionsCount();
+  int output_shape_DimensionsCount =
+      tflite::micro::GetTensorShape(output).DimensionsCount();
+  int* output_shape_DimsData =
+      const_cast<int*>(tflite::micro::GetTensorShape(output).DimsData());
+
+  void* params = (void*)&op_params;
+  int8_t* inputp =
+      const_cast<int8_t*>(tflite::micro::GetTensorData<int8_t>(input));
+  int8_t* filterp =
+      const_cast<int8_t*>(tflite::micro::GetTensorData<int8_t>(filter));
+  int32_t* biasp =
+      const_cast<int32_t*>(tflite::micro::GetTensorData<int32_t>(bias));
+  int8_t* outputp =
+      const_cast<int8_t*>(tflite::micro::GetTensorData<int8_t>(output));
+
+#ifdef MCPS_MEASUREMENT
+  int batches = output_shape_DimsData[0];
+  int output_depth = weights_shape_DimsData[weights_shape_DimensionsCount - 2];
+  int accum_depth = weights_shape_DimsData[weights_shape_DimensionsCount - 1];
+  MCPS_START_ONE;
+#endif
+
+  int sizeof_scr = output_shape_DimsData[1];
+
+  if (sizeof_scr > CEVA_TFLM_KERNELS_SCRATCH_SIZE_VAL) {
+    TF_LITE_KERNEL_LOG(context, "Scratch size (%d) less that required (%d)",
+                       CEVA_TFLM_KERNELS_SCRATCH_SIZE_VAL, sizeof_scr);

```

Comment:
With https://github.com/tensorflow/tensorflow/pull/46904, you can not directly call MicroPrintf instead of TF_LITE_KERNEL_LOG(context